hoplite-watch/src/main/kotlin/com/sksamuel/hoplite/watch/watchers/file.kt

```diff
@@ -0,0 +1,46 @@
+package com.sksamuel.hoplite.watch.watchers
+
+import com.sksamuel.hoplite.watch.Watchable
+import java.nio.file.FileSystems
+import java.nio.file.Paths
+import java.nio.file.StandardWatchEventKinds
+import java.util.concurrent.Executors
+
+class FileWatcher(private val dir: String) : Watchable {
+  var cb = {}
+
+  override fun watch(callback: () -> Unit) {
+    cb = callback
+    start()
+  }
+
+  private fun start() {
+    val watchService = FileSystems.getDefault().newWatchService()
+    val pathToWatch = Paths.get(dir).toAbsolutePath()
+
+    pathToWatch.register(
+      watchService,
+      StandardWatchEventKinds.ENTRY_CREATE,
+      StandardWatchEventKinds.ENTRY_MODIFY,
+      StandardWatchEventKinds.ENTRY_DELETE
+    )
+
+    Executors.newSingleThreadExecutor().submit {
+      while (true) {
+        val watchKey = watchService.take()
```

Comment:
Does this block. If not we will want a sleep in the loop

Reply:
The docs say `Retrieves and removes next watch key, waiting if none are yet present.` which makes me think it does block.

Reply:
Sounds like it.
Put everything inside the while (true) in a try catch block. .take() can throw if an interrupt fires and we don't want the while loop to exit if so.